### ISSUE TITLE
which function return meDiscard?

### ISSUE THREAD

**Issue by modelica-trac-importer** [tbeu/fmi-standard#4]
**Reported by mauss on 4 Oct 2009 13:37 UTC**
A program that uses an FMU must catch all cases where a function returns meDiscard. Currently, every function may return meDiscard. For example: what step to discard if meSetTime returns meDiscard? Should a FMU processor check for this case?
I propose to restrict meDiscard return value to only those functions where discard makes sense, to simplify implementation of FMU processors.

----

Migrated-From: https://trac.fmi-standard.org/ticket/4

**Comment by modelica-trac-importer** [comment]
**Modified by Jakob Mauss on 4 Oct 2009 13:43 UTC**

**Comment by modelica-trac-importer** [comment]
**Modified by andreas on 25 Nov 2009 20:06 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by Torsten Blochwitz on 4 Dec 2009 10:45 UTC**
I think the following functions could return me_Discard:
meGetReal, meGetDerivatives, meGetIndicators, meCompletedIntegrationStep 

A critical point here is the following possible calling sequence (e.g. of an explicit integration method):
- meSetTime
- meSetReal
- meCompletedIntegrationStep
...

If a calling environment has to deal with more than one FMUs it calls meCompletedIntegrationStep for each of them. If the last FMU returns meDiscard, the others may have already done their actions for an accepted step and the environmebt is not able to repeat the step (e.g. with smaller stepsize).

I am not sure if it makes sense that meEventUpdate returns meDiscard. If meEventUpdate can not return meDiscard it makes no sense that meGetInteger, Boolean and String can return meDiscard, because these values may change only after meEventUpdate is called.

**Comment by modelica-trac-importer** [comment]
**Comment by Martin Otter on 6 Dec 2009 17:29 UTC**
I added a sentence to meDiscard:

 meDiscard: this return status is only possible, if explicitly defined for the corresponding function (currently: meGetReal, meGetDerivatives, meGetIndicators): ...

and added a note to

 meGetReal, meGetDerivatives, meGetIndicators 

as suggested by Torsten. I did not add it to meCompletedIntegrationStep, because the integration step is completed and it is then not possible to reduce the step size any more. If this would be needed, the model has to return meError.

**Comment by modelica-trac-importer** [comment]
**Modified by dietmarw on 7 Sep 2012 10:18 UTC**
